### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by petarcurkovic**
@d4be4st @DamirSvrtan 
I made some changes related to this issue #9 

Feel free to suggest some improvements :)

### CODE REVIEW

**Comment by d4be4st** [comment]
@petarcurkovic how would you define custom instance methods on Enumeration::Value ?

**Comment by petarcurkovic** [comment]
Hi @d4be4st !

I totally forgot about custom instance methods! 

I've fixed this now.
Now there is a module `Enumerations::Value` which contains all instance methods for single enumeration.
This module is included into `Enumerations::Base`.

**Comment by d4be4st** [comment]
Since you are on it, @petarcurkovic you can add scopes similar to rails enum.

User.role_admins to return all users with role :admin

**Comment by petarcurkovic** [comment]
Hi @d4be4st !

It's done.

File: lib/enumerations/value.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    # Example:
+    #
+    #   Role.admin => #<Enumeration::Value:0x007fff45d7ec30 @base=Role, @symbol=:admin,
+    #                   @attributes={:id=>1, :name=>"Admin", :description=>"Some description..."}>
+    #   user.role.id => # 1
+    #   user.role.name => # "Admin"
+    #   user.role.description => # "Some description..."
+    #
+    def create_instance_methods
+      @attributes.each do |key, _|
```
Comment (d4be4st) [inline lib/enumerations/value.rb]: move this into its own method as you did with `define_value_checking_method`

**Pull request merged**